### ISSUE TITLE
Formatter can insert mysterious extraneous parentheses in `with` statements

### ISSUE THREAD

**Issue by lpulley** [astral-sh/ruff#14001]
My `ruff --version` is 0.7.1.

I've encountered a case where `ruff format` adds nesting parentheses for seemingly no reason:

```py
with (
    open(
        "some/path.txt",
        "rb",
    )
    if True
    else open("other/path.txt")
    # Bar
):
    pass
```

```
➜ ruff format --diff up034.py
--- up034.py
+++ up034.py
@@ -1,10 +1,12 @@
 with (
-    open(
-        "some/path.txt",
-        "rb",
+    (
+        open(
+            "some/path.txt",
+            "rb",
+        )
+        if True
+        else open("other/path.txt")
     )
-    if True
-    else open("other/path.txt")
     # Bar
 ):
     pass

1 file would be reformatted
```

i.e. it wants this:

```py
with (
    (
        open(
            "some/path.txt",
            "rb",
        )
        if True
        else open("other/path.txt")
    )
    # Bar
):
    pass
```

UP034 complains about this, for good reason.

If the `# Bar` comment is removed, the formatter makes no changes. More curiously, if I keep `# Bar` but add `as _`, the formatter makes no changes. (I also checked that Black does not do this.)

I've resolved the issue in the meantime by moving the `# Bar` comment to a different line, but this seems like a formatter bug.

**Comment by MichaReiser** [comment]
Thanks for reporting this. Agree, this looks silly but I'm a bit scared from fixing it... with item formatting is complicated.

**Comment by lpulley** [comment]
Hmm... with Ruff 0.7.2, the example I gave above is still problematic; it still wants to add extra `(` `)` and fail UP034.

**Comment by MichaReiser** [comment]
> Hmm... with Ruff 0.7.2, the example I gave above is still problematic; it still wants to add extra `(` `)` and fail UP034.

Yeah, we can't fix it before promoting the new style to stable. You would have to use `format.preview = true` for now

**Comment by lpulley** [comment]
Aha; I missed that. Thanks for the clarification!!

-------- Original Message --------

**Comment by MichaReiser** [comment]
You can subscribe to #13371 if you want to know when the preview style gets stabilized.

**Comment by lpulley** [comment]
I'm trying out 0.9.0 and still seeing this behavior. Did this make it into the 2025 style?

**Comment by MichaReiser** [comment]
Hmmm, no. That's on me.

**Comment by lpulley** [comment]
Not the end of the world; just wanted to make sure I wasn't missing anything obvious. Thanks.

**Comment by MichaReiser** [comment]
Thanks for the kind words. 

We might want to pull it into 0.10, considering that changes should be super rare. I don't feel comfortable shipping it as part of a patch release because it would violate our versioning policy

**Comment by cj81499** [comment]
Sounds good to me, thank you!

**Comment by MichaReiser** [comment]
This fix is now merged into the release branch for 0.10
